core/domain/collection_jobs_one_off.py

```diff
@@ -96,3 +96,82 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class RemoveCollectionRightsTranslatorIdsOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that sets the translator_ids in *CollectionRightsModel to None
+    in order to remove it from the datastore.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(RemoveCollectionRightsTranslatorIdsOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            collection_models.CollectionRightsModel
+        ]
+
+    @staticmethod
+    def map(collection_rights_model):
+        # This is the only way to remove the field from the model,
+        # see https://stackoverflow.com/a/15116016/3688189 and
+        # https://stackoverflow.com/a/12701172/3688189.
+        if 'translator_ids' in collection_rights_model._properties:  # pylint: disable=protected-access
+            del collection_rights_model._properties['translator_ids']  # pylint: disable=protected-access
```

Comment:
Since now we delete it in the `convert_to_valid_dict` we should do `del model_dict['translator_ids']` instead of `model_dict['translator_ids'] = []`